OpenRA.Game/Map/MapPreview.cs

```diff
@@ -272,47 +276,81 @@ bool EvaluateUserFriendliness(Dictionary<string, PlayerReference> players)
 			return true;
 		}
 
-		public void UpdateRemoteSearch(MapStatus status, MiniYaml yaml)
+		public void UpdateRemoteSearch(MapStatus status, MiniYaml yaml, Action<MapPreview> parseMetadata = null)
 		{
-			// Update on the main thread to ensure consistency
-			Game.RunAfterTick(() =>
+			if (status == MapStatus.DownloadAvailable)
 			{
-				if (status == MapStatus.DownloadAvailable)
+				// HACK/PERF: The following is very much not thread safe for code (mainly UI) that uses the
+				// properties. We can get away without locking because the outside code must always check the
+				// Status field, and knows not to access it unless it is Available or DownloadAvailable.
+				// This means that no other threads should be accessing these info fields until *after* the
+				// Status field is set to MapStatus.DownloadAvailable at the end of the method.
```

Comment:
The UI thread picks up the changes from the `Status` field changing.  That's what I was trying to explain with the comment.